lib/normalize.js

```diff
@@ -32,6 +33,16 @@ var _splitLocator = function (nemo, locatorString) {
 module.exports = function normalize(nemo, _locator) {
   var locator = _locator;
   var normalizedLocator;
+
+  if (locator.using && locator.value && nemo.wd.By[locator.using]) {
+    normalizedLocator = nemo.wd.By[locator.using](locator.value);
+    if(normalizedLocator.constructor === locator.constructor){
```

Comment:
Having success with this version of `normalize`:

```js
module.exports = function normalize(nemo, _locator) {
  var locator = _locator;
  var normalizedLocator;

  if (locator.constructor !== String && !locator.using && !locator.value) {
    if (!locator.locator || !locator.locator.trim()) {
      //throw new Error('NO or EMPTY locator found, please fix it');
    }
    if (!locator.type || !locator.type.trim() || !nemo.wd.By[locator.type]) {
      //throw new Error('NO or EMPTY or invalid locator Type found, please fix it');
    }
  }
  if (locator.using && locator.value && nemo.wd.By[locator.using]) {
    return locator;
  }
  else if (_locator.constructor === String) {
    locator = _splitLocator(nemo, _locator);
  }
  normalizedLocator = nemo.wd.By[locator.type](locator.locator);
  return normalizedLocator;
};
```

I don't like the error conditions exactly, but it satisfies all of the new test cases.